cpp/src/arrow/dataset/file_base.h

```diff
@@ -386,7 +389,7 @@ struct ARROW_DS_EXPORT FileSystemDatasetWriteOptions {
   /// and only write the row groups to the disk when sufficient rows have accumulated.
   /// The final row group size may be less than this value and other options such as
   /// `max_open_files` or `max_rows_per_file` lead to smaller row group sizes.
-  uint64_t min_rows_per_group = 0;
+  uint64_t min_rows_per_group = 10;
```

Comment:
Why 10 and not something larger? Alternatively, why set this by default at all?

Reply:
If partitioning, it is very possible to end up with tiny row groups.  For example, if we partition by year, and a batch comes in with 1 million rows spread across 1000 years you would end up with 1000 row groups with 1000 rows which is undesirable.

However, the default here should be 1 << 20 (1Mi)

Reply:
According with the ticket https://issues.apache.org/jira/browse/ARROW-15409?filter=-1, it suggest to set a value higher than 0, I use the 10 as I saw in some tests, what value would be apropiate for this?
@westonpace the `max_rows_per_group` is already `1 << 20` maybe I need to put a lower value

Reply:
It's fine for `max_rows_per_group` and `min_rows_per_group` to have the same value I think.  `1 << 18` is probably ok.  The unit tests used `10` because I needed to be able to test the various features without generating a whole bunch of data (which would be time consuming) but `10` would have poor performance in practice because that means we would need to write a big block of metadata every 10 rows.